src/formElementPlugins/checkbox/src/ApplicationView.tsx

```diff
@@ -199,14 +256,54 @@ export const getCheckboxStructure = (
         selected: cb?.[checkboxPropertyNames.selected] || false,
       }
   })
+
+  if (hasOther)
+    checkboxElements.push({
+      label: otherLabel,
+      text: initValues ? initValues.other?.text ?? '' : '',
+      textNegative: '',
+      key: 'other',
+      selected: false,
+    })
+
   // On first render, we set elements to the *saved* state, but after that we
-  // replace them with the new, changed checkbox values
-  return isFirstRender
-    ? checkboxElements.map((cb) => ({
-        ...cb,
-        selected: initValues?.[cb.key] ? initValues[cb.key].selected : cb.selected,
-      }))
-    : checkboxElements
+  // replace them with the new, changed checkbox values, or default values
+  if (isFirstRender && initValues)
+    return checkboxElements.map((cb) => ({
+      ...cb,
+      selected: initValues?.[cb.key] ? initValues[cb.key].selected : cb.selected,
+    }))
+
+  if (defaultValue) return standardiseDefault(defaultValue, checkboxElements, hasOther)
+
+  return checkboxElements
+}
+
+const standardiseDefault = (
+  defaultValue: string | { [key: string]: Checkbox } | Checkbox[],
+  checkboxElements: Checkbox[],
+  hasOther: boolean
+): Checkbox[] => {
+  if (typeof defaultValue === 'string') {
+    // Convert delimited text string to selection values
+    let textValues = defaultValue.split(',').map((e) => e.trim())
+    const newCheckboxes = checkboxElements.map((cb) => {
+      const newCheckbox = textValues.includes(cb.text)
+        ? { ...cb, selected: true }
+        : { ...cb, selected: false }
+      textValues = textValues.filter((e) => e !== cb.text)
+      return newCheckbox
+    })
+    if (hasOther && textValues.length > 0) {
+      // Leftover textValues must be for the "Other" value
+      newCheckboxes[newCheckboxes.length - 1].text = textValues[0]
+      newCheckboxes[newCheckboxes.length - 1].selected = true
```

Comment:
I would have like to have used the common `useDefault` hook for default handling, but because this element has its own dynamically handling of changing checkbox values, they wouldn't really play nicely together.

So the default handling is kind of unique to this element, the main feature being that it can uses a comma-delimited string to populate checkboxes (such as is needed for the company "sub-type" field).

Reply:
From a configurer's perspective, I would have thought an array of strings would be more intuitive given that is how the options are set - I wonder if this is easy to implement, it might be helpful for setting config?

Reply:
Yeah, good point. The reason I have it this way is because of wanting to prefill values from the database. For example, the "company subtype" output is saved as a comma-seperated string, so in order to pre-fill these checkboxes for company edit, we need it to be able to handle it this way.

You're right though that it makes more sense for the default format to be the same as the actual checkboxes. I might see if I can get it to handle both. I'll let you know if/when I've done it.

Reply:
Okay @fergie-nz, I've made a small change to allow an array of strings as default, too. d1be5e46